server/index.js

```diff
@@ -89,25 +89,112 @@ function getCurrentFixtures(round, currentSeason, leagueId) {
     .then(result => {
-      const dbresult = result.rows[0];
-      const todayGames = dbresult.fixtures;
-      const time = new Date();
-      const formatDate = dateFns.format(time, 'yyyy-MM-dd');
-      const filter = todayGames.filter(fixture => {
-        return fixture.fixture.date.slice(0, 10) === formatDate;
+      const dbresult = result.rows;
+      const todayGames = dbresult.map(fixtures => {
+        const filteredGames = fixtures.fixtures.filter(fixture => {
+          const time = new Date();
+          const formatDate = dateFns.format(time, 'yyyy-MM-dd');
+          return fixture.fixture.date.slice(0, 10) === formatDate;
+        });
+        return filteredGames;
       });
-      res.json(filter);
+      const flatten = todayGames.flat(1);
+      res.json(flatten);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/team-form/:teamId', (req, res, next) => {
+  const { utcDate } = req.body.teamId;
+  const sql = `
+    select "matchDetails"
+    from "teamForm"
+    where "date" = $1
+  `;
+  const params = [utcDate];
+  db.query(sql, params)
+    .then(result => {
+      if (result.rows.length) {
+        return result.rows;
+      }
+      return Promise.all([
+        getHomeForm(req.body.teamId),
+        getAwayForm(req.body.teamId)
+      ])
+        .then(concatPromises => {
+          return [].concat.apply([], concatPromises);
+        })
+        .then(matchDetails => {
+          const jsonMatchDetails = JSON.stringify(matchDetails);
+          const sql = `
+      insert into "teamForm" ("date", "matchDetails")
+      values ($1, $2)
+      returning "matchDetails"
+      `;
+          const params = [utcDate, jsonMatchDetails];
+          return db.query(sql, params).then(result => {
+            return result.rows[0];
+          })
+            .then(matchdetails => {
+              return res.json(matchdetails);
+            });
+        });
+    }).then(matchDetails => {
+      return res.json(matchDetails);
     })
     .catch(err => next(err));
 });
 
+function getAwayForm(obj) {
+  const init = {
+    method: 'GET',
+    url: 'https://api-football-v1.p.rapidapi.com/v3/teams/statistics',
+    params: { league: obj.leagueId, season: obj.currentSeason, team: obj.awayId, date: obj.date },
+    headers: {
+      'x-rapidapi-key': process.env.API_FOOTBALL_API_KEY,
+      'x-rapidapi-host': 'api-football-v1.p.rapidapi.com'
+    }
+  };
+  return axios.request(init).then(response => {
+    return response.data.response;
+  });
+}
+function getHomeForm(obj) {
+  const init = {
+    method: 'GET',
+    url: 'https://api-football-v1.p.rapidapi.com/v3/teams/statistics',
+    params: { league: obj.leagueId, season: obj.currentSeason, team: obj.homeId, date: obj.date },
+    headers: {
+      'x-rapidapi-key': process.env.API_FOOTBALL_API_KEY,
+      'x-rapidapi-host': 'api-football-v1.p.rapidapi.com'
+    }
+  };
+  return axios.request(init).then(response => {
+    return response.data.response;
+  });
+}
```

Comment:
These functions are identical except one needs a `homeId` and the other needs an `awayId`. Just make one function that uses a `teamId`.


```js
function getTeamStats(leagueId, season, teamId, date) { 
  const init = {   
    method: 'GET',  
    url: 'https://api-football-v1.p.rapidapi.com/v3/teams/statistics',   
    params: { league: leagueId, season: season, team: teamId, date: date },  
    headers: {
      'x-rapidapi-key': process.env.API_FOOTBALL_API_KEY,
      'x-rapidapi-host': 'api-football-v1.p.rapidapi.com'
    }
  }; 
  return axios.request(init).then(response => {   
    return response.data.response;
  });
}
```